### DISCUSSION TITLE
default save history for Database revision

### DISCUSSION THREAD

**Discussion by pavlis** [mspass-team/mspass#449] [Design & Development]
This is a minor issue for the new Database implementation, but worth preserving for the record.  As you know I've been cleaning up duplicate code in the `Database` class.  In the process as noted elsewhere I've been pushing to deprecate the functions that explicitly tag ensembles and simply reads and writes to two core method of Database:  `save_data` and `read_data`.  

With that background a simple question comes up.  The current version of `Database` will always save any object-level history stored in a `TimeSeries` or `Seismogram` within the `ProcessingHistory` container.  The user has no choice.   If the container has data it will be saved.   I made a choice to add a new boolean argument to `save_data` with the obvious name `save_history`.   The function is backward compatible with the current version of `save_data` when I set the default to True.   Several tests failed when I initially set the default False.  From the above that would be expected, but I didn't remember that feature of the current implementation and the errors were baffling for a while.   

So there are two points I want to preserve here for the record:
1.  Do you, @wangyinz , concur that adding this save_history options makes sense? 
2. If so, should the default remain True?   Since we advocate history as something to use only for preserving a final workflow to promote reproducible science, maybe the default should be False?   If so the several tests that broke with that selection will need some repair.

**Comment by wangyinz** [comment]
yes, we should provide that option, which will mirror the one in read_data:
https://github.com/mspass-team/mspass/blob/23898f45753603606c529b0034e3bac7eaf80ac2/python/mspasspy/db/database.py#L337

I think keeping the default as true is probably better. It wouldn't hurt to save the history even if users don't want it, but it will hurt in the opposite.
